### PR TITLE
Use versioned libblastrampoline and GMP

### PR DESCRIPTION
**Pull request by gbaraldi**
Basically the title.

### CODE REVIEW

**Review by staticfloat — APPROVED** [review]

**Comment by staticfloat** [comment]
Very interesting; both win32 and win64 died with approximately the same error.  Any ideas as to why?

**Comment by staticfloat** [comment]
It's minimally reproducible with the following:

```
julia> using LinearAlgebra, SparseArrays
       using SparseArrays: fixed
       b = sprandn(10, 10, 0.99) + I
       a = fixed(b)
       lu(a) \ randn(10)

Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x0 -- unknown function (ip: 0000000000000000)
in expression starting at REPL[6]:5
unknown function (ip: 0000000000000000)
Allocations: 1954934 (Pool: 1953138; Big: 1796); GC: 3
```

**Comment by gbaraldi** [comment]
I imagine it's the blas trampoline change, because the GMP change works in the mimalloc PR.

**Comment by giordano** [comment]
> It's minimally reproducible with the following:

https://github.com/JuliaLinearAlgebra/libblastrampoline/issues/89

**Comment by staticfloat** [comment]
That looks quite probable:
```
julia> println.(filter(x -> occursin("blast", x), Libdl.dllist()))
C:\Users\Administrator\AppData\Local\Temp\julia-0f526b01f8\bin\libblastrampoline-5.dll
C:\Users\Administrator\AppData\Local\Temp\julia-0f526b01f8\bin\libblastrampoline.dll
```

**Comment by giordano** [comment]
So there is still something pulling in the wrong library.

**Comment by staticfloat** [comment]
It's SuiteSparse:

```
$ for f in *.dll; do if [[ -n $(objdump.exe -p $f | grep "libblastrampoline.dll") ]]; then echo $f; fi; done
libblastrampoline.dll
libcholmod.dll
libspqr.dll
libumfpack.dll
```

File: base/gmp.jl [hunk 1]
```diff
@@ -21,8 +21,16 @@ else
 end
 const CdoubleMax = Union{Float16, Float32, Float64}
 
-version() = VersionNumber(unsafe_string(unsafe_load(cglobal((:__gmp_version, :libgmp), Ptr{Cchar}))))
-bits_per_limb() = Int(unsafe_load(cglobal((:__gmp_bits_per_limb, :libgmp), Cint)))
+if Sys.iswindows()
+    const libgmp = "libgmp-10.dll"
```
Comment (vchuravy) [inline base/gmp.jl:25]: Can we pull in the version from the buildsystem? See base/Makefile the build_h.jl generation

**Comment by vchuravy** [comment]
One minor nit we should move the so version into a place where packagers can override them, and it would be great to just have a single place to change them

File: base/gmp.jl (hunk 1 above)
Comment (giordano) [reply base/gmp.jl:25]: @gbaraldi bump

Comment (gbaraldi) [reply base/gmp.jl:25]: I'm not sure, maybe we can pull the version from the jll from inside the makefile? It's kind of annoying not being able to depend on the jll proper here.

Comment (gbaraldi) [reply base/gmp.jl:25]: In fact, is there a reason for BigNums being part of base instead of being stdlibs?

Comment (giordano) [reply base/gmp.jl:25]: They're used in Base, no?

Comment (staticfloat) [reply base/gmp.jl:25]: They’re needed quite early because we can parse literal bignums

Comment (staticfloat) [reply base/gmp.jl:25]: We can get version numbers from the makefile system, then embed those into build_h.jl. I suggest we try some awk trickery to figure out the so major version number.

**Comment by giordano** [comment]
SuiteSparse is still failing on Windows despite the update to the new build 😕

**Comment by giordano** [comment]
```console
sandbox:${WORKSPACE} # for f in bin/*.dll; do if [[ -n $(${target}-objdump -p $f | grep "libblastrampoline.dll") ]]; then echo $f; fi; done
bin/libblastrampoline.dll
sandbox:${WORKSPACE} # for f in bin/*.dll; do if [[ -n $(${target}-objdump -p $f | grep "libblastrampoline-5.dll") ]]; then echo $f; fi; done
bin/libblastrampoline-5.dll
sandbox:${WORKSPACE} # for f in bin/*.dll; do if [[ -n $(${target}-objdump -p $f | grep "libblastrampoline-5-2-0.dll") ]]; then echo $f; fi; done
bin/libblastrampoline-5-2-0.dll
bin/libblastrampoline-5.dll
bin/libblastrampoline.dll
bin/libcholmod.dll
bin/libspqr.dll
bin/libumfpack.dll
```

So now we're targeting `libblastrampoline-5-2-0.dll`, which is even worse than `libblastrampoline.dll`

**Comment by giordano** [comment]
All Windows tests are green 🥳

**Pull request merged**